src/Validation/Validator.php

```diff
@@ -635,8 +706,92 @@ public function requirePresence($field, $mode = true, $message = null)
+     *
+     * You can also set flags, when and message for all passed fields, the individual
+     * setting takes precedence over group settings.
+     *
+     * ### Example:
+     *
+     * ```
+     * // Email can be empty
+     * $validator->allowEmptyByFlags('email', Validator::EMPTY_STRING);
+     *
+     * // Email can be empty on create
+     * $validator->allowEmptyByFlags('email', Validator::EMPTY_STRING, 'create');
+     *
+     * // Email can be empty on update
+     * $validator->allowEmptyByFlags('email', Validator::EMPTY_STRING, 'update');
+     *
+     * // Email and subject can be empty on update
+     * $validator->allowEmptyByFlags(['email', 'subject'], Validator::EMPTY_STRING, 'update');
+     *
+     * // Email can be always empty, subject and content can be empty on update.
+     * $validator->allowEmptyByFlags(
+     *      [
+     *          'email' => [
+     *              'when' => true
+     *          ],
+     *          'content' => [
+     *              'message' => 'Content cannot be empty'
+     *          ],
+     *          'subject'
+     *      ],
+     *      Validator::EMPTY_STRING,
+     *      'update'
+     * );
+     * ```
+     *
+     * It is possible to conditionally allow emptiness on a field by passing a callback
+     * as a second argument. The callback will receive the validation context array as
+     * argument:
+     *
+     * ```
+     * $validator->allowEmpty('email', Validator::EMPTY_STRING, function ($context) {
+     *   return !$context['newRecord'] || $context['data']['role'] === 'admin';
+     * });
+     * ```
+     *
+     * If you want to allow other kind of empty data on a field, you need to pass other
+     * flags:
+     *
+     * ```
+     * $validator->allowEmptyByFlags('photo', Validator::EMPTY_FILE);
+     * $validator->allowEmptyByFlags('published', Validator::EMPTY_STRING | Validator::EMPTY_DATE | Validator::EMPTY_TIME);
+     * $validator->allowEmptyByFlags('items', Validator::EMPTY_STRING | Validator::EMPTY_ARRAY);
+     * ```
+     *
+     * You can also use convenience wrappers of this method. The following calls are the
+     * same as above:
+     *
+     * ```
+     * $validator->allowEmptyFile('photo');
+     * $validator->allowEmptyDateTime('published');
+     * $validator->allowEmptyArray('items');
+     * ```
+     *
+     * @param string|array $field the name of the field or a list of fields
+     * @param int|null $flags A bitmask of EMPTY_* flags which specify what is empty
+     * @param bool|string|callable $when Indicates when the field is allowed to be empty
+     * Valid values are true (always), 'create', 'update'. If a callable is passed then
+     * the field will allowed to be empty only when the callback returns true.
+     * @param string|null $message The message to show if the field is not
+     * @return $this
+     */
+    public function allowEmptyByFlags($field, $flags, $when = true, $message = null)
```

Comment:
Does this need to be public? It is a complicated to operate method that has simpler to use public methods defined below.

Reply:
As I said in the dev channel, people might need to customize what is empty. For example, if it is a RESTful API, people may prefer EMPTY_ARRAY only.

Reply:
Ok.